CI-CD/Jenkins/Mend CLI/vars/MendSCAScan.groovy

```diff
@@ -1,17 +1,32 @@
-def call() { 
-      echo 'Run Mend dependencies scan'
-      catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE'){
-            sh '''
-            export repo=$(basename -s .git $(git config --get remote.origin.url))
-            export branch=$(git rev-parse --abbrev-ref HEAD)
-            ./mend dep -u -s "*//${JOB_NAME}//${repo}_${branch}" --fail-policy --non-interactive --export-results dep-results.txt
-            if [[ "$dep_exit" == "9" ]]; then
-                  echo "[warning]  Dependency scan policy violation"
+def call(Map args = [:]) { 
+    boolean reachability = args.get('reachability', false)
+    echo 'Run Mend dependencies scan'
+    catchError(buildResult: 'SUCCESS', stageResult: 'FAILURE') {
+        if (reachability) {
```

Comment:
This if statement should only be for the ``mend dep`` command. Everything else should only be in the script once.